### PR TITLE
Improve dirty grid rendering performance.

### PR DESCRIPTION
**Pull request by ZehMatt**
Software (HW), Multi-threading off (1080p):
- OpenRCT2 Title sequence (first park):
-- develop: 50~ FPS
-- PR: 80~ FPS
- bpb:
-- develop: 28~ FPS
-- PR: 45~ FPS

Software (HW), Multi-threading on (1080p):
- OpenRCT2 Title sequence (first park):
-- develop: 145~ FPS
-- PR: 155~ FPS
- bpb:
-- develop: 90~ FPS
-- PR: 95~ FPS

Clearly some gains with multi-threading off, this is also something OpenLoco apparently does.

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
I remember we changed this in OpenLoco to fix a bug, even. See https://github.com/OpenLoco/OpenLoco/issues/744 and https://github.com/OpenLoco/OpenLoco/pull/761

**Comment by Gymnasiast** [comment]
Is it okay if I take this for a test?

File: src/openrct2/drawing/X8DrawingEngine.cpp [hunk 1]
```diff
@@ -374,18 +393,10 @@ void X8DrawingEngine::DrawAllDirtyBlocks()
...
-            // Determine columns
-            uint32_t xx;
-            for (xx = x; xx < _dirtyGrid.BlockColumns; xx++)
-            {
-                if (_dirtyGrid.Blocks[yOffset + xx] == 0)
-                {
-                    break;
-                }
-            }
+            // See comment above to why this is 1.
```
Comment (Gymnasiast) [inline src/openrct2/drawing/X8DrawingEngine.cpp:396]:
```suggestion
            // See comment above as to why this is 1.
```

**Comment by ZehMatt** [comment]
> Is it okay if I test this out?

ofc

**Comment by Gymnasiast** [comment]
With optimisations (again BPB at 2560 × 1440):

Without multithreading:
Develop: 30 FPS
PR: 38 FPS

With multithreading:
Develop: 50 FPS
PR: 56 FPS

**Comment by janisozaur** [comment]
Looks like it's worthy of a changelog entry

**Comment by ZehMatt** [comment]
> With optimisations (again BPB at 2560 × 1440):
> 
> Without multithreading: Develop: 30 FPS PR: 38 FPS
> 
> With multithreading: Develop: 50 FPS PR: 56 FPS

Thanks, we are almost back to 40 FPS (vanilla) at 2k so that's not bad I guess. Would still feel a lot better at 60+ or even 120 when one owns a high refresh rate monitor :eyes:

**Comment by duncanspumpkin** [comment]
Awe man i remember working for ages to work out this issue in openloco but that was to fix z fighting. I tried it out in openrct2 but as it didn't fix z fighting i didn't proceed any further with porting it.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**